Code-the-Dream-School/cc-prac-team4-front
# PP-109: implement filters functionality on the pets page

**Filters functionality on the pets page implemented**

**Pull Request Description:**

This pull request implements the filters functionality on the pets page. The changes include updates to the petAction.js, PetCard.js, and Pets.js files.

The petAction.js file now includes a new function getSearchPetFilters that handles the search filters for pets. It takes various parameters such as petType, breed, age, size, gender, goodWith, coatLength, color, careAndBehaviour, and dispatch. The function makes an API call to retrieve the filtered pet results and dispatches the appropriate actions.

In the PetCard.js file, there is a modification to display the age as "Kitten" for cats and "Puppy" for dogs when the age is "Baby."

The Pets.js file contains several changes. It introduces a new reducer SearchPetFiltersReducer to handle the state related to pet filters. The component also utilizes new state variables and handlers for selecting filter options such as selectedType, selectedBreed, selectedAge, selectedSize, selectedGender, selectedGoodWith, selectedCoatLength, selectedColor, and selectedCareAndBehav. When any filter option is changed, it calls the petFiltersResults function to update the pet filters results based on the selected options. Additionally, the setPets state is used to store the filtered pets to be displayed on the page.

Added a "Clear all filters" button from React Bootstrap library to the Pets component. This button allows users to easily reset all selected filters and view all available pets.

**Testing Steps:**

1. Go to the pets page [http://localhost:3000/pets](http://localhost:3000/pets) and verify that the filters functionality is working as expected.
2. Test different filter combinations and verify that the pet results are updated accordingly.
3. Check the console for any errors or warnings.
4. Test the modified PetCard.js behavior by verifying that the age is displayed as "Kitten" for cats and "Puppy" for dogs when the age is "Baby".
5. Verify that the pagination functionality is still working correctly.
6. Test the API calls made by the getSearchPetFilters function and ensure that the correct data is fetched and displayed on the page.
7. Verify that the "Clear all filters" button is visible and positioned correctly within the filter navigation section.
8. Click the "Clear all filters" button and confirm that all filters are reset to their default state.
9. Apply different filters again and verify that the changes are reflected in the pet results.
10. Test the button's functionality by clicking it multiple times and ensuring that the filters reset consistently.

<img width="1403" alt="Screenshot 2023-06-26 at 2 53 09 AM" src="https://github.com/Code-the-Dream-School/cc-prac-team4-front/assets/2952900/616a4317-948d-4665-86ac-fa96861dcbd5">
<img width="1425" alt="Screenshot 2023-06-26 at 2 53 59 AM" src="https://github.com/Code-the-Dream-School/cc-prac-team4-front/assets/2952900/d259d38d-8a8b-4744-9b64-25e607dfcc70">
